app-drawer/app-drawer.html

```diff
@@ -645,10 +637,16 @@
       },
 
       _openedPersistentChanged: function() {
-        if (this.transitionDuration === 0) {
-          // Reset drawer state now since there will be no transitionend event.
-          this._resetDrawerState();
+        // Don't reset drawer state if tracking or flinging, since _trackEnd and _flingDrawer will
+        // set its own timer.
+        if (this._drawerState === this._DRAWER_STATE.FLINGING ||
+            this._drawerState === this._DRAWER_STATE.TRACKING) {
+          return;
         }
+
+        // Use setTimeout instead of transitionend since transitionend won't fire when app-drawer
+        // is display: none.
+        window.setTimeout(this._resetDrawerState.bind(this), this.transitionDuration);
```

Comment:
Consider using `debounce('animation', fn, waitTime)` here.  That way, you are safe that only one timer will fire. You can also use `Polymer.dom.flush()` in the tests to avoid those nested timeOuts. It might also be posible to remove the condition in line 642.